src/util.js

```diff
@@ -1,12 +1,16 @@
-export function _normalizeAbsolutePath(path, hasPushState) {
+export function _normalizeAbsolutePath(path, hasPushState, absolute = false) {
   if (!hasPushState && path[0] !== '#') {
     path = '#' + path;
   }
 
+  if (hasPushState && absolute) {
+    path = path.substring(1, path.length);
+  }
+
   return path;
 }
 
-export function _createRootedPath(fragment, baseUrl, hasPushState) {
+export function _createRootedPath(fragment, baseUrl, hasPushState, absolute) {
```

Comment:
It seems like we should be able to eliminate the `hasPushState` hacks now that we know the base URL, right?